### PR TITLE
expr: fix tikv crash when conv empty string

### PR DESCRIPTION
**Pull request by wshwsh12**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/12673

What's Changed:

```commit-message
Fix tikv crash when conv empty string
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Fix tikv crash when conv empty string
```

### CODE REVIEW

File: components/tidb_query_expr/src/impl_math.rs [hunk 1]
```diff
@@ -616,6 +616,9 @@ fn is_valid_base(base: IntWithSign) -> bool {
 
 fn extract_num_str(s: &str, from_base: IntWithSign) -> Option<(String, bool)> {
     let mut iter = s.chars().peekable();
+    if iter.peek().is_none() {
+        return None;
+    }
     let head = *iter.peek().unwrap();
```
Comment (sticnarf) [inline components/tidb_query_expr/src/impl_math.rs:622]: It can be simplify it as:
```suggestion
    let head = *iter.peek()?;
```

**Review by youjiali1995 — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Pull request merged**